sdks/java/harness/src/main/java/org/apache/beam/fn/harness/WindowMergingFnRunner.java

```diff
@@ -154,7 +155,13 @@ public void merge(Collection<W> toBeMerged, W mergeResult) throws Exception {
       for (KV<W, Collection<W>> mergedWindow : mergedWindows) {
         currentWindows.removeAll(mergedWindow.getValue());
       }
-      return KV.of(windowsToMerge.getKey(), KV.of(currentWindows, (Iterable) mergedWindows));
+      KV result =
```

Comment:
Can we state the type specially here? And please squash all commits into one then this PR is ready to go.